searchcore/src/vespa/searchcore/proton/bucketdb/checksumaggregators.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "checksumaggregators.h"
+#include <xxhash.h>
+
+namespace proton::bucketdb {
+
+using BucketChecksum = ChecksumAggregator::BucketChecksum;
+using Timestamp = ChecksumAggregator::Timestamp;
+using GlobalId = ChecksumAggregator::GlobalId;
+
+namespace {
+
+uint32_t
+gidChecksum(const document::GlobalId &gid)
+{
+    union {
+        const unsigned char *_c;
+        const uint32_t *_i;
+    } u;
+    u._c = gid.get();
+    const uint32_t *i = u._i;
+    return i[0] + i[1] + i[2];
```

Comment:
Strictly speaking I think this kind of union type aliasing is undefined in C++ (but not C). `memcpy`'ing parts of the GID into `uint32_t`s would be a well-defined alternative and also probably fewer lines in total.

Reply:
Fixed